pkg/loki/modules.go

```diff
@@ -312,9 +312,12 @@ func (t *Loki) initQueryFrontend() (_ services.Service, err error) {
 
 	return services.NewIdleService(nil, func(_ error) error {
 		t.frontend.Close()
-		if t.stopper != nil {
-			t.stopper.Stop()
+		for _, stopper := range t.stoppers {
+			if stopper != nil {
+				stopper.Stop()
```

Comment:
Is this correct? calling `Stop` sequentially is fine or should this be called in go routine?

Reply:
Sounds fine to me.